lizard_auth_server/models.py

```diff
@@ -745,6 +749,44 @@ def as_dict(self):
 # New models
 # ##########
 
+class ProfileManager(models.Manager):
+    def get_queryset(self):
+        qs = super(ProfileManager, self).get_queryset()
+        try:
+            user = request.user
+        except RuntimeError:
+            return qs
+        if user.is_superuser:
+            return qs
+        # If the user is admin of one or more companies return all the users
+        # belonging to those companies.
+        # NOTE 1: this means an admin is able to edit users that belong to a
+        # company other than his or her own!
+        # NOTE 2: to be able to manage your own company's users you have to
+        # be explicitly admin of your company (which is in line with the
+        # implementation).
+        return qs.filter(company__in=user.profile.companies_as_admin.all())
+
+
+class CompanyManager(models.Manager):
+    def get_queryset(self):
+        qs = super(CompanyManager, self).get_queryset()
+        try:
+            user = request.user
+        except RuntimeError:
+            return qs
+        if user.is_superuser:
+            return qs
+        # Return your own company; if you're admin or guest at a company,
+        # return those companies as well.
+        company_pk = user.profile.company.pk if user.profile.company else None
+        companies_as_admin = qs.filter(
+            Q(administrators__pk__contains=user.profile.pk) |
+            Q(guests__pk__contains=user.profile.pk) |
+            Q(pk=company_pk))
+        return companies_as_admin
```

Comment:
Hoi Reinout, ik twijfel eigenlijk of ik uberhaupt Company moet filteren. De reden dat ik bijv. guests erbij heb staan is dat anders de has_access methode faalt, want als je inlogt als een normale gebruiker (zonder admin of superrechten), dan ga worden ook allerlei queries uitgevoerd die nu gefiltered worden door de speciale object manager. Dus volgens mij is het best tricky als je dit soort dingen doet om ik hiervoor overal ervan uitging dat ik gewoon alle objecten kon krijgen.